src/clients/c++/perf_client/perf_utils.h

```diff
@@ -112,3 +117,32 @@ std::string GetRandomString(const int string_length);
 template <Distribution distribution>
 std::function<std::chrono::nanoseconds(std::mt19937&)> ScheduleDistribution(
     const double request_rate);
+
+// Create a shared memory region of the size 'byte_size' and return the unique
+// identifier.
+// \param shm_key The string identifier of the shared memory region
+// \param byte_size The size in bytes of the shared memory region
+// \param shm_id Returns an int descriptor of the created shared memory region
+// \return error Returns if an error if unable to open shared memory region.
+nic::Error CreateSharedMemoryRegion(
+    std::string shm_key, size_t byte_size, int* shm_fd);
+
+// Mmap the shared memory region with the given 'offset' and 'byte_size' and
+// return the base address of the region.
+// \param shm_id The int descriptor of the created shared memory region
+// \param offset The offset of the shared memory block from the start of the
+// shared memory region
+// \param byte_size The size in bytes of the shared memory region
+// \param shm_addr Returns the base address of the shared memory region
+// \return error Returns if an error if unable to mmap shared memory region.
+nic::Error MapSharedMemory(
+    int shm_fd, size_t offset, size_t byte_size, void** shm_addr);
+
+// Destory the shared memory region with the given name.
+// \return error Returns if an error if unable to unlink shared memory region.
+nic::Error UnlinkSharedMemoryRegion(std::string shm_key);
+
+// Munmap the shared memory region from the base address with the given
+// byte_size.
+// \return error Returns if an error if unable to unmap shared memory region.
+nic::Error UnmapSharedMemory(void* shm_addr, size_t byte_size);
```

Comment:
I think @CoderHam has extracted these functions as common utilities for client library, you probably can just use that directly.

Reply:
Yes, refer to `examples/shm_utils.h` (Only system shared memory)

Reply:
Done